tests/python/unittest/test_gluon_data.py

```diff
@@ -158,10 +158,11 @@ def __getitem__(self, key):
 def test_multi_worker():
     data = Dataset()
     for thread_pool in [True, False]:
-        loader = gluon.data.DataLoader(data, batch_size=1, num_workers=5, thread_pool=thread_pool)
-        for i, batch in enumerate(loader):
-            assert (batch.asnumpy() == i).all()
-
+        for auto_reload in [True, False]:
```

Comment:
Use pytest.mark.parametrize instead of a for loop

Reply:
why and how...
Actually I just do not know how to use `pytest.mark.parametrize`.
and, I saw such code at line 263:
```
    for epoch in range(1):
        for i, data in enumerate(loader):
```
it seems that a `for-loop` is acceptable even if there is only one parameter in the iterator.

Reply:
parametrize is used to parametrize different variations of the test function. Please read the doc https://docs.pytest.org/en/stable/parametrize.html
It's necessary to ensure parallelization on our CI works smoothly (by avoiding giant test functions with nested for loops)

Reply:
Here you should use
```
@pytest.mark.parametrize("auto_reload", [True, False])
@pytest.mark.parametrize("thread_pool", [True, False])
```

instead of nested for loop

Reply:
After a second thought, I'm not sure if pytest will work correctly on the v1.x branch. So if the test fails now, you may need to revert back to your previous implementation. Sorry, the v1.x is legacy branch and many of the bugs and infrastructure issues are only fixed in the master branch.